### PR TITLE
Add support for space group information when reading and writing pdb files

### PR DESCRIPTION
**Pull request by alex-sbaq**
Added `get_space_group()` and `set_space_group()` methods as discussed in #689 to support space group and Z value reading/writing for (deprecated) PDB files.

### CODE REVIEW

**Review by padix-key — CHANGES_REQUESTED** [review]
Hi, thanks for implementing this functionality! Could you add a test as well to `tests/structure/io/test_pdb.py`? I think something analogous to `test_conversion()` would work well here: Setting and subsequent retrieving of the space group should return the original space group.

**Comment by padix-key** [comment]
Thanks for the update! I will give the PR a final review on the weekend.

File: tests/structure/io/test_pdb.py [hunk 1]
```diff
@@ -77,6 +77,51 @@ def test_array_conversion(path, model, hybrid36, include_bonds):
...
+        # If parsing fails for model=None due to mismatched atom counts, skip the test
+        if model is None:
+            pytest.skip("Skipping test due to incompatible atom counts across models.")
+        else:
+            raise  # Re-raise the error for other cases
+
+    # Write the structure and space group back to a new PDB file
+    pdb_file = pdb.PDBFile()
+    pdb.set_structure(pdb_file, stack1)
+    pdb.PDBFile.set_space_group(pdb_file, cryst1)
```
Comment (padix-key) [inline tests/structure/io/test_pdb.py:114]: I think this is more concise and idiomatic.

```suggestion
    pdb_file.set_space_group(cryst1)
```

File: tests/structure/io/test_pdb.py [hunk 2]
```diff
@@ -77,6 +77,51 @@ def test_array_conversion(path, model, hybrid36, include_bonds):
     assert array1.coord.tolist() == array2.coord.tolist()
 
 
+@pytest.mark.parametrize(
+    "path, model",
+    itertools.product(glob.glob(join(data_dir("structure"), "*.pdb")), [None, 1, -1]),
```
Comment (padix-key) [inline tests/structure/io/test_pdb.py:82]: The number of parsed structure models does not influence the `CRYST1` record. Hence for the sake of test speed, I propose to remove the `model` parametrization from this test and always use `model=1` instead.

**Review by padix-key — CHANGES_REQUESTED** [review]
Thanks for adding the test. I have some minor requests for it, but otherwise, everything looks good!

File: tests/structure/io/test_pdb.py (hunk 1 above)
Comment (padix-key) [inline tests/structure/io/test_pdb.py:114]: I think this is more concise and idiomatic.

```suggestion
    pdb_file.set_structure(stack1)
    pdb_file.set_space_group(cryst1)
```

**Comment by alex-sbaq** [comment]
Thanks for the feedback, @padix-key! I've made the changes to the test - let me know if anything is missing.

**Review by padix-key — APPROVED** [review]
Thanks, the PR looks good to me. I will just wait for the CI to see if the tests pass

**Pull request merged**